### PR TITLE
Labs

### PR DESCRIPTION
**Pull request by ltonello**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
added code that completes labs from week 1

## What did you learn from the changes you have made?
learned how to work with models,  multi layer neural networks and recommender systems

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Nope

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Nope

## How were these changes tested?
All code runs

## Checklist
- [X] I can confirm that my changes are working as intended

### CODE REVIEW

File: 01_materials/labs/lab_1.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_1.ipynb:1]: - Feed Forward Neural Networks with Keras:
  - c: Please always reinitialize your model with model = Sequential() ... to reset all model parameters. Or, you are continuing the training from the previous trained state with a different model setup. You may check this with the accuracy of the first epoch.

File: 01_materials/labs/lab_2.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_2.ipynb:1]: - Hyper parameters settings: Please experiment with a few hyper parameters of the `NeuralNet` model.

**Review by tianyi21 — APPROVED** [review]
Hi Lauren!

Thank you for submitting your DL assignments. Please refer to my comments attached to each file.

Your Lab 1-3 are now complete.

Thanks,
Tianyi [LS]

File: 01_materials/labs/lab_4.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_4.ipynb:1]: - A simple convolution filter: You defined a `Conv2D` with kernel size 20 but initialized it with kernel size 5 (my_kernel)
- Using a pretrained model: I personally suggest pre-processing the image with `preprocess_input()` without `resize()` when using a pretrained model, since it may introduce artifacts to the image and lead to strange predictions. `preprocessing_input()` will resize the image automatically.

File: 01_materials/labs/lab_5.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_5.ipynb:1]: - Build a better model: Typically we don't use activation functions in the `box_prediction_head` just like what is given in [Classification and Localization model], since we're predicting parameters defining bounding boxes, although `relu` is acceptable here.

**Review by tianyi21 — APPROVED** [review]
Hi Lauren,

Thank you for submitting your DL assignments. Please refer to my comments attached to each file.

Your Lab 4-6 are now complete.

Thanks,
Tianyi [LS]
